src/Raven.Server/Documents/Handlers/Debugging/QueriesDebugHandler.cs

```diff
@@ -93,5 +96,143 @@ public Task RunningQueries()
 
             return Task.CompletedTask;
         }
+
+        private void WritePropertyNameAndScalarValue(BlittableJsonTextWriter writer, string propertyName,object propertyValue)
+        {
+            writer.WritePropertyName(propertyName);
+
+            if (propertyValue is Int64 || propertyValue is Int32 || propertyValue is Int16 ||
+                propertyValue is UInt64 || propertyValue is UInt32 || propertyValue is UInt16 ||
+                propertyValue is byte || propertyValue is sbyte)
+            {
+                writer.WriteValue(BlittableJsonToken.Integer, (long)propertyValue);
+            }
+            else if (propertyValue is LazyStringValue lsv)
+            {
+                writer.WriteString(lsv);
+            }
+            else if (propertyValue is string str)
+            {
+                writer.WriteString(str);
+            }
+            else
+            {
+                writer.WriteString(propertyValue.ToString());
+            }            
+            
+        }
+       
+        [RavenAction("/databases/*/debug/queries/cache/list", "GET", AuthorizationStatus.ValidUser, IsDebugInformationEndpoint = true)]
+        public Task QueriesCacheList()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var queryCache = Database.QueryMetadataCache.GetQueryCache();
+
+                var responseDjv = new DynamicJsonValue();
+                var queriesList = new List<DynamicJsonValue>();
+
+                writer.WritePropertyName("TotalCachedQueries");
+                writer.WriteValue(BlittableJsonToken.Integer,(long)queryCache.Length);
```

Comment:
I think we have a `writer.WriteNumber` or `write.WriteInteger` for this. No need to do this like that.

Reply:
yeah, correct, it's write integer